src/daemon/dlt_daemon_socket.c

```diff
@@ -224,6 +228,14 @@ int dlt_daemon_socket_sendreliable(int sock, void *data_buffer, int message_size
         if (ret < 0) {
             dlt_vlog(LOG_WARNING,
                      "%s: socket send failed [errno: %d]!\n", __func__, errno);
+#ifdef DLT_SYSTEMD_WATCHDOG_ENABLE
+            /* notify systemd here that we are still alive
+             * otherwise we might miss notifying the watchdog when
+             * the watchdog interval is small and multiple timeouts occur back to back
+             */
+            if (sd_notify(0, "WATCHDOG=1") < 0)
+                dlt_vlog(LOG_WARNING, "%s: Could not reset systemd watchdog\n", __func__);
```

Comment:
Is there a reason why dlt_daemon_trigger_systemd_watchdog_if_necessary() was not used here?

Reply:
Because there is no timestamp to use as a reference time. We could call the function and always pass 0 as timestamp. 
As you mentioned in the other comment this patch improves the watchdog behavior but it's far from perfect. 
Especially in environment with congested networks between the clients and the daemon combined with heavy logging load.

Reply:
Got it. Thanks.